tests/components/melcloud/test_config_flow.py

```diff
@@ -0,0 +1,204 @@
+        return_value=mock_coro(True),
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {"email": "test-email@test-domain.com", "password": "test-password"},
+        )
+
+    assert len(mock_login.mock_calls) == 1
+    assert result2["type"] == "abort"
+    assert result2["reason"] == reason
+
+
+@pytest.mark.parametrize(
+    "error,message",
+    [(401, "invalid_auth"), (403, "invalid_auth"), (500, "cannot_connect")],
+)
+async def test_form_response_errors(
+    hass, mock_login, mock_get_devices, mock_request_info, error, message
+):
+    """Test we handle response errors."""
+    mock_login.return_value = mock_coro(
+        exception=ClientResponseError(mock_request_info(), (), status=error),
+    )
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with async_patch(
+        "homeassistant.components.melcloud.async_setup", return_value=mock_coro(True)
+    ), async_patch(
+        "homeassistant.components.melcloud.async_setup_entry",
+        return_value=mock_coro(True),
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {"email": "test-email@test-domain.com", "password": "test-password"},
+        )
+
+    assert result2["type"] == "abort"
+    assert result2["reason"] == message
+
+
+async def test_failed_import_form(hass, mock_login, mock_get_devices):
+    """Test we get the form if imported without token."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data={},
+    )
+    assert result["type"] == "form"
+    assert result["errors"] is None
+
+
+async def test_import_with_token(hass, mock_login, mock_get_devices):
+    """Test successful import."""
+    with async_patch(
+        "homeassistant.components.melcloud.async_setup", return_value=mock_coro(True)
+    ) as mock_setup, async_patch(
+        "homeassistant.components.melcloud.async_setup_entry",
+        return_value=mock_coro(True),
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={"email": "test-email@test-domain.com", "token": "test-token"},
+        )
+
+    assert result["type"] == "create_entry"
+    assert result["title"] == "test-email@test-domain.com"
+    assert result["data"] == {
+        "email": "test-email@test-domain.com",
+        "token": "test-token",
+    }
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_token_refresh(hass, mock_login, mock_get_devices):
+    """Re-configuration with existing email should refresh token."""
+    await hass.config_entries.async_add(
+        config_entries.ConfigEntry(
```

Comment:
Use `MockConfigEntry` imported from tests/common.py.

Reply:
Neat! Switched to the mock version.